Sources/TuistKit/Generator/TuistGeneratorModelLoader.swift

```diff
@@ -0,0 +1,215 @@
+import Basic
+import Foundation
+import TuistCore
+import TuistGenerator
+
+enum TuistGeneratorModelLoaderError: Error, FatalError {
+    case malformedManifest
+    
+    var type: ErrorType {
+        switch self {
+        case .malformedManifest:
+            return .bug
+        }
+    }
+    
+    var description: String {
+        switch self {
+        case .malformedManifest:
+            return "The Project manifest appears to be malformed"
+        }
+    }
+}
+
+class TuistGeneratorModelLoader: ModelLoading {
+    
+    private let fileHandler: FileHandling
+    private let manifestLoader: GraphManifestLoading
+    
+    init(fileHandler: FileHandling, manifestLoader: GraphManifestLoading) {
+        self.fileHandler = fileHandler
+        self.manifestLoader = manifestLoader
+    }
+    
+    func loadProject(at path: AbsolutePath) throws -> Project {
+        let json = try manifestLoader.load(.project, path: path)
+        let project = try TuistKit.Project.from(json: json, path: path, fileHandler: fileHandler)
+        return project
+    }
+    
+    func loadWorkspace(at path: AbsolutePath) throws -> Workspace {
+        let json = try manifestLoader.load(.workspace, path: path)
+        let workspace = try TuistKit.Workspace.from(json: json, path: path)
+        return workspace
+    }
+}
+
+extension TuistKit.Workspace {
+    static func from(json: JSON, path: AbsolutePath) throws -> TuistKit.Workspace {
+        let projectsStrings: [String] = try json.get("projects")
+        let name: String = try json.get("name")
+        let projectsRelativePaths: [RelativePath] = projectsStrings.map { RelativePath($0) }
+        let projects = projectsRelativePaths.map { path.appending($0) }
+        return Workspace(name: name, projects: projects)
+    }
+}
+
+extension TuistKit.Project {
+    static func from(json: JSON, path: AbsolutePath, fileHandler: FileHandling) throws -> TuistKit.Project {
+        let name: String = try json.get("name")
+        let targetsJSONs: [JSON] = try json.get("targets")
+        let targets = try targetsJSONs.map { try TuistKit.Target.from(json: $0, path: path, fileHandler: fileHandler) }
+        let settingsJSON: JSON? = try? json.get("settings")
+        let settings = try settingsJSON.map { try TuistKit.Settings.from(json: $0, path: path, fileHandler: fileHandler) }
+        
+        return Project(path: path,
+                       name: name,
+                       settings: settings,
+                       targets: targets)
+    }
+}
+
+extension TuistKit.Target {
+    static func from(json: JSON, path: AbsolutePath, fileHandler: FileHandling) throws -> TuistKit.Target {
+        let name: String = try json.get("name")
+        let platform = TuistKit.Platform(rawValue: try json.get("platform"))!
+        let product = TuistKit.Product(rawValue: try json.get("product"))!
```

Comment:
Is the force unwrap here required? Could we throw a descriptive error instead?

Reply:
aah missed this! too much copying from our POC branch :p - will push an update